src/test/kotlin/io/github/fstaudt/hugo/TestProject.kt

```diff
@@ -52,11 +52,20 @@ fun TestProject.runAndFail(vararg task: String): BuildResult {
     return gradleRunner(*task).buildAndFail()
 }
 
-private fun TestProject.gradleRunner(vararg task: String): GradleRunner {
-    return GradleRunner.create()
-        .withProjectDir(this)
-        .withArguments("--info", "--stacktrace", *task)
-        .withPluginClasspath()
-        .withDebug(true)
-        .forwardOutput()
-}
+private fun TestProject.gradleRunner(vararg task: String): GradleRunner =
+    GradleRunner.create().withProjectDir(this).apply {
+        withArguments(
+            "--info",
+            "--stacktrace",
+            "--warning-mode=fail",
+            "--configuration-cache",
+            *task
+        )
+        withPluginClasspath()
+        withDebug(true)
+        forwardOutput()
+        gradleVersionSystemProp?.let<String, Unit> { withGradleVersion(it) }
+    }
+
+internal val gradleVersionSystemProp: String?
+    get() = System.getProperty("testGradleVersion")
```

Comment:
gradleVersionSystemProp doesn't need to be nullable as it is always set by Gradle build.

An inlined function named `testGradleVersion()` would be more clear than a variable.
```suggestion
internal fun testGradleVersion(): String = System.getProperty("testGradleVersion")
```

Reply:
Done